integration-tests/src/tests/client/sandbox.rs

```diff
@@ -122,36 +122,68 @@ fn test_patch_account() {
 #[test]
 #[cfg(feature = "sandbox")]
 fn test_fast_forward() {
+    const BLOCKS_TO_FASTFORWARD: BlockHeight = 10_000;
+    const BLOCKS_TO_PRODUCE: u64 = 20;
+
+    // the 1_000_000 is to convert milliseconds to nanoseconds, where MIN/MAX_BLOCK_PROD_TIME are in milliseconds:
+    const FAST_FORWARD_DELTA: u64 = (BLOCKS_TO_FASTFORWARD + BLOCKS_TO_PRODUCE) * 1_000_000;
+
     init_test_logger();
     run_actix(async {
         let count = Arc::new(AtomicUsize::new(0));
+        let first_block_timestamp = Arc::new(AtomicU64::new(0));
+
         // Produce 20 blocks
-        let (client, _view_client) = setup_mock(
+        let (_client, _view_client) = setup_mock(
             vec!["test".parse().unwrap()],
             "test".parse().unwrap(),
             true,
             false,
-            Box::new(move |msg, _ctx, _| {
+            Box::new(move |msg, _ctx, client| {
                 if let NetworkRequests::Block { block } = msg.as_network_requests_ref() {
                     let height = block.header().height();
+                    let timestamp = block.header().raw_timestamp();
+
                     count.fetch_add(1, Ordering::Relaxed);
-                    if count.load(Ordering::Relaxed) >= 20 {
+                    let at = count.load(Ordering::Relaxed);
+
+                    // Fast forward by 10,000 blocks after the first block produced:
+                    if at == 1 {
+                        first_block_timestamp.store(timestamp, Ordering::Relaxed);
+                        client.do_send(NetworkClientMessages::Sandbox(
+                            NetworkSandboxMessage::SandboxFastForward(10000),
+                        ));
+                    }
+
+                    // Check at final block if timestamp and block height matches up:
+                    if at >= BLOCKS_TO_PRODUCE as usize {
+                        let before_forward = first_block_timestamp.load(Ordering::Relaxed);
+                        let min_forwarded_time =
+                            before_forward + FAST_FORWARD_DELTA * MIN_BLOCK_PROD_TIME;
+                        let max_forwarded_time =
+                            before_forward + FAST_FORWARD_DELTA * MAX_BLOCK_PROD_TIME;
                         assert!(
-                            height >= 10000,
+                            height >= BLOCKS_TO_FASTFORWARD,
                             "Was not able to fast forward. Current height: {}",
                             height
                         );
+
+                        // Check if final block timestamp is within fast forwarded time min and max:
+                        assert!(
+                            (min_forwarded_time..max_forwarded_time).contains(&timestamp),
+                            "Forwarded timestamp {} is not within expected range ({},{})",
+                            timestamp,
+                            min_forwarded_time,
+                            max_forwarded_time
+                        );
+
```

Comment:
also check epoch height is bumped here?

Reply:
Epoch height isn't available unfortunately within integration-tests since that is provided by the `EpochManager` crate. Not sure how I could tie that in since epoch manager isn't being tested with that the rest of nearcore here. And it seems like epoch manager crate was meant to be separate anyways since I see docs saying it could potentially be turned into a contract

Reply:
Actually, was able to test epoch height in `fast_forward.py` instead since those tests have everything end-to-end